connectors/katalog/pkg/connector/connector.go

```diff
@@ -57,3 +72,90 @@ func (r *Handler) getAssetInfo(c *gin.Context) {
 
 	c.JSON(http.StatusOK, &response)
 }
+
+func (r *Handler) reportError(c *gin.Context, httpCode int, errorMessage string) {
+	r.log.Error().Msg(errorMessage)
+	c.JSON(httpCode, gin.H{"error": errorMessage})
+}
+
+// Enables writing of assets to katalog. The different flows supported are:
+// (a) When DestinationAssetID is specified:
+//     Then a destination asset id is created with name : <DestinationAssetID>
+// (b) When DestinationAssetID is not specified but ResourceMetadata.Name of source asset is specified:
+//     Then an asset is created with name: <ResourceMetadata.Name>-<Kubernetes Generated Random String>
+// (c) When DestinationAssetID and ResourceMetadata.Name of source asset are not specified:
+//     Then an asset is created with name: fybrik-<Kubernetes Generated Random String>
+func (r *Handler) createAssetInfo(c *gin.Context) {
+	// Parse request
+	var request datacatalog.CreateAssetRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
+		r.reportError(c, http.StatusBadRequest, "Error during ShouldBindJSON in createAssetInfo"+err.Error())
+		return
+	}
+
+	logging.LogStructure("CreateAssetRequest object received:", request, &r.log, false, false)
+
+	if request.DestinationCatalogID == "" {
+		r.reportError(c, http.StatusBadRequest, "Invalid DestinationCatalogID in request")
+		return
+	}
+
+	var asset *v1alpha1.Asset
+	var err error
+	assetNamePrefix := FybrikAssetPrefix
+	if request.ResourceMetadata.Name != "" {
+		assetNamePrefix = request.ResourceMetadata.Name + "-"
+	}
+
+	asset = &v1alpha1.Asset{
+		ObjectMeta: v1.ObjectMeta{Namespace: request.DestinationCatalogID, Name: request.DestinationAssetID, GenerateName: assetNamePrefix},
+		Spec: v1alpha1.AssetSpec{
+			SecretRef: v1alpha1.SecretRef{Name: request.Credentials},
+			Metadata:  request.ResourceMetadata,
+			Details:   request.Details,
+		},
+	}
+	if request.DestinationAssetID != "" {
+		// scenario when asset name is not generated by kubernetes
+		// update the name field in the resourcemetadata to be asset name for consistency
+		// this step ensures we dont call controller runtime update method later and hence we can save a method call.
+		asset.Spec.Metadata.Name = request.DestinationAssetID
+	}
+
+	logging.LogStructure("Fybrik Asset to be created in Katalog:", asset, &r.log, false, false)
+
+	err = r.client.Create(context.Background(), asset)
+	if err != nil {
+		r.reportError(c, http.StatusInternalServerError, "Error during create asset. Error:"+err.Error())
+		return
+	}
+	logging.LogStructure("Created Asset: ", asset, &r.log, false, false)
+
+	// scenario when asset name is generated by kubernetes with appropriate assetPrefix
+	// update the name field in the resourcemetadata by calling controller runtime update method to be asset name for consistency
+	if request.DestinationAssetID == "" {
+		asset.Spec.Metadata.Name = asset.ObjectMeta.Name
+
+		err := wait.Poll(PollPeriod*time.Millisecond, PollTimeout*time.Second, func() (bool, error) {
```

Comment:
you missed the Get call here. The idea of reupdate that if the destination value was changed, you have to get the latest value before the next update. lat's talk tomorrow.